src/main/scala/dpla/ingestion3/harvesters/api/CdlHarvester.scala

```diff
@@ -0,0 +1,83 @@
+
+package dpla.ingestion3.harvesters.api
+
+import java.net.URI
+
+import org.apache.commons.io.IOUtils
+import org.apache.http.client.methods.{CloseableHttpResponse, HttpGet}
+import org.apache.http.client.utils.URIBuilder
+import org.apache.http.impl.client.HttpClients
+import org.apache.http.util.EntityUtils
+import scala.util.{Failure, Success, Try}
+
+class CdlHarvester(val endpoint: String,
+                   val queryParams: Map[String, String] = Map()) {
+
+  /**
+    * Get a single-page, unparsed response from the OAI feed, or an error if
+    * one occurs.
+    *
+    * @param start The pagination offset
+    * @return OaiSource or OaiError
+    */
+  def harvest(start: String = "0"): CdlResponse = {
+    getUri(queryParams.updated("start", start)) match {
+      // Error building URL
+      case Failure(e) =>
+        val source = CdlSource(queryParams)
+        CdlError(e.toString, source)
+      case Success(url) => {
+        getJsonResponse(url) match {
+          case Failure(e) =>
+            CdlError(e.toString, CdlSource(queryParams, Some(url.toString)))
+          case Success(response) => response.isEmpty match {
+            case true => CdlError("Response body is empty", CdlSource(queryParams, Some(url.toString)))
+            case false => CdlSource(queryParams, Some(url.toString), Some(response))
+          }
+        }
+      }
+    }
+  }
+
+  /**
+    * Builds query URI from parameters
+    *
+    * @param queryParams
+    * @return URI
+    */
+  def getUri(queryParams: Map[String, String]): Try[URI] = {
+
+    Try {
+      new URIBuilder()
+        .setScheme("https")
+        .setHost(endpoint)
+        .setPath("/solr/query")
+        .setParameter("q", queryParams.getOrElse("query", "*:*"))
+        .setParameter("start", queryParams.getOrElse("start", "0"))
+        .setParameter("rows", queryParams.getOrElse("rows", "10"))
+        .build()
+      }
+    }
+
+  /**
+    * Makes request and returns JSON response
+    *
+    * @param uri
+    * @return
+    */
+  def getJsonResponse(uri: URI): Try[String] = {
+    Try {
+      val httpclient = HttpClients.createDefault()
+      val get = new HttpGet(uri)
+      get.addHeader("X-Authentication-Token", queryParams.getOrElse("api_key", ""))
+      var response: CloseableHttpResponse = null
+      try {
+        response = httpclient.execute(get)
+        val entity = response.getEntity
+        EntityUtils.toString(entity)
+      } finally {
+        IOUtils.closeQuietly(response)
+      }
```

Comment:
Do you really need three layers of nested `Try` / `try` blocks here?  It seems at first glance that at least the second `Try` is superfluous.

Reply:
Looks like he's ensuring the closeQuietly method gets called, which is different than managing errors.